src/byte-helpers.js

```diff
@@ -0,0 +1,39 @@
+export const isTypedArray = (obj) => ArrayBuffer.isView(obj);
+export const toUint8 = (bytes) => (bytes instanceof Uint8Array) ? bytes :
+  new Uint8Array(bytes && bytes.buffer || bytes, bytes && bytes.byteOffset || 0, bytes && bytes.byteLength || 0);
+
+export const bytesToString = (bytes) => {
+  if (!bytes) {
+    return '';
+  }
+
+  bytes = Array.prototype.slice.call(bytes);
+
+  // TODO: when we remove ie 11 support, check if TextDecoder has better performance
+
+  const string = String.fromCharCode.apply(null, toUint8(bytes));
+
+  try {
+    return decodeURIComponent(escape(string));
+  } catch (e) {
+    // if decodeURIComponent/escape fails, we are dealing with non string data,
```

Comment:
In that case string will usually be a garbled mess of characters, perhaps we should return empty string in that case, but that could be problematic because it would make using this function a bit harder in some cases. 

Example 1: Where returning a partially garbled string may help

You think the string `ID3` starts a byte 0 and ends at byte 4, while really it ends at byte 3.
You call this function with a subarray of 0 to 4. `string` will be `ID3~` where tilde is a garbled character. if we returned an empty string here you wouldn't really know what is going on.

Example 2: Where returning a partially garbled string doesn't help, but doesn't really hurt either.

You think the string `竜竜` starts a byte 0 and ends at byte 5, while really it ends at byte 4 (since non-ascii characters codes are greater than 255). 
You call this function with a subarray of 0 to 5. `string` will be entirely garbled `~#~#)`. Regardless of what we return it is confusing, but even the garbled charactes could help you determine that the character codes are off as `~#` in google may lead you to someone who tells you that that is actually character 387 which is `竜` and from there you know your off by one byte.